framework/core/js/src/common/Application.tsx

```diff
@@ -262,7 +263,22 @@ export default class Application {
   }
 
   public boot() {
-    this.initializers.toArray().forEach((initializer) => initializer(this));
+    const caughtInitializationErrors: CallableFunction[] = [];
+
+    this.initializers.toArray().forEach((initializer) => {
+      try {
+        initializer(this);
+      } catch (e) {
```

Comment:
Shouldn't we rethrow `e` so the original error is accessible?

Reply:
The error is still accessible since we display it with `console.error` in the introduced helper. (see screenshot above)
If we throw the execution halts and the app will display the nojs content:

https://github.com/flarum/framework/blob/b64003cba5ae77575babccc1803773d12fe5e2f7/framework/core/views/frontend/app.blade.php#L27-L36

Reply:
Makes sense!